### DISCUSSION TITLE
Set exact=true as default in invalidateQueries()

### DISCUSSION THREAD

**Discussion by chrisurlaub** [TanStack/query#3014] [General] [answered]
Hi,

I do not understand why the QueryFilters-prop "exact" in invalidateQueries() does not have "true" as default.
With regard to https://github.com/tannerlinsley/react-query/issues/1725, this actually seems to be in your sense (?).

With the default "false" the option leads to more requests than necessary and sometimes also to unexpected behaviour.

Alternatively, it would be OK for me if the option could be set globally via the QueryClientProvider.

Thanks!

**Comment by TkDodo** [comment, marked as answer]
#1725 only wanted `exact` as `true` for `getQueryData`, which makes total sense. Otherwise, you might get a random entry (the first one that matches), which is not what is wanted.

For invalidation, I think it all boils down to how you structure your query keys: https://tkdodo.eu/blog/effective-react-query-keys

if you have:

```
["todos"]
["todos", "1"]
["todos", "2"]
```

and you only want `todos`, you need to specify exact. Which is why I prefer:

```
["todos", { type: list }]
["todos", { type: 'detail', id: 1 }]
["todos", { type: 'detail', id: 2 }]
```

because that very explicitly allows me to either target everything todo related, or only the list (or lists, if I have multiple with filters), or a specific detail, without needing the exact option.

**Reply by chrisurlaub** [reply]
Thx a lot for the quick reply!
